openmc/bounding_box.py

```diff
@@ -99,19 +107,58 @@ def extend(self, padding_distance: float) -> BoundingBox:
 
         Returns
         -------
-        An enlarged bounding box
+        An expanded bounding box
+        """
+        self[0] -= padding_distance
+        self[1] += padding_distance
+        return self
+
+    def expand(self, other_box: BoundingBox, inplace: bool = False) -> BoundingBox:
+        """Expand the box to contain another box
+
+        Parameters
+        ----------
+        other_box : BoundingBox
+            The box used to resize this box
+        inplace : bool
+            Whether or not to return a new BoundingBox instance or to modify the
+            coefficients of this plane.
+
+        Returns
+        -------
+        An expanded bounding box
+        """
+        self[0] = np.minimum(self.lower_left, other_box.lower_left)
+        self[1] = np.maximum(self.upper_right, other_box.upper_right)
+        return self if inplace else BoundingBox(*self)
```

Comment:
The implication of having `inplace=False` is that you _don't_ change the original object. Here it would modify the original object _and_ return a new copy.